src/modinfo.c

```diff
@@ -1242,16 +1242,41 @@ sr_oper_data_merge_cb(struct lyd_node *trg_node, const struct lyd_node *src_node
     return LY_SUCCESS;
 }
 
+/**
+ * @brief Use (and spend) the oper push data cache if available for the module.
+ *
+ * @param[in] sess Session to use.
+ * @param[in] module_name Module name to get the cache for.
+ * @return The oper push data cache of the module, if available.
```

Comment:
I would say you still cannot distinguish no cached data and empty cached data. Best to just use 2 out parameters `data` and `found`, for example. And use `void` return value since the function cannot fail.

Reply:
I don't think this is necessary. 
```C
         if (!oper_push_dup[i].has_data) {
              /* no push oper data */
              continue;
          }
```
is checked in `sr_module_oper_data_load()` just before calling `sr_module_oper_data_use_cache()`.
So, there is no ambiguity.
If `sr_module_oper_data_use_cache()` returns `NULL`, we must always load the data from the DS plugin callback.

Reply:
I see, a bit confusing. Can that actually happen, that there are no cached push oper data but there are some stored?